### PR TITLE
Some real magic

### PR DESCRIPTION
**Pull request by raed667**
shit happens here ----- >

### CODE REVIEW

File: content/overlay.js [hunk 1]
```diff
@@ -15,7 +15,32 @@ var FakeAddress = {
...
 
+	this.Mailauthor = this.fullAuthor.slice(n+1,m);
+	
+	var at = this.Mailauthor.indexOf("@");
+
+	this.Mailerauthor = this.Mailauthor.slice(at+1,this.Mailauthor.length);
+	
+	delete window.n;
+	delete window.m;
+
```
Comment (rednaks) [inline content/overlay.js:34]: So from L18 to L33 you are doing the same thing as me from L43 to L46 !? Except that I'm not storing it as a class member.

File: content/overlay.js [hunk 2]
```diff
@@ -15,7 +15,32 @@ var FakeAddress = {
...
+	var at = this.Mailauthor.indexOf("@");
+
+	this.Mailerauthor = this.Mailauthor.slice(at+1,this.Mailauthor.length);
+	
+	delete window.n;
+	delete window.m;
+	
+	var at = this.messageID.indexOf("@");	
+	this.MailermessageID = this.messageID.slice(at+1,this.messageID.length);
+
```
Comment (rednaks) [inline content/overlay.js:37]: But this showed me that I'm doing it wrong at line 104 when I'm comparing `this.messageID` with the host's message-id

File: content/overlay.js [hunk 3]
```diff
@@ -76,8 +101,28 @@ var FakeAddress = {
   {
     var hostname = config.incoming.hostname;
     if(hostname === 'imap.googlemail.com'){
-      if(this.messageID !== 'mail.google.com')
-        alert('Warning, this might be a fake email !\nThe message-id is : "'+this.messageID+'" instead of "<someRandomAlphanumericCharacters>@'+hostname);
-    }
+      //if(this.messageID !== 'mail.google.com')
+      //  alert('Warning, this might be a fake email !\nThe message-id is : "'+this.messageID+'" instead of "<someRandomAlphanumericCharacters>@'+hostname);
+        if(this.Mailerauthor === 'esprit.tn' /*esprit.tn to be replaced with a var checked from the database*/)
```
Comment (rednaks) [inline content/overlay.js]: We have to think about the way we can store the white list. We have some cases that a domain might have many message-ids, this is the case of google services. 
What about opening an issue discussion for this ?

**Comment by rednaks** [comment]
Okey, I'll keep this PR open if you can fix the code :)

File: content/overlay.js (hunk 3 above)
Comment (raed667) [inline content/overlay.js]: next commit will fix this in an array for each domain 

so expl .. i check esprit.tn

i get the array ("esprit.tn","mail.google.com","mail.gmail.com")

File: content/overlay.js (hunk 1 above)
Comment (raed667) [inline content/overlay.js:34]: I will  need that later ... + I do it with style 8)

**Comment by raed667** [comment]
for now it works perfectly with esprit ... waiting for a database to add another array for the hostname

File: content/overlay.js (hunk 1 above)
Comment (raed667) [inline content/overlay.js:34]: I stick by my code !

**Pull request merged**